### PR TITLE
shell-app-system: obey X-Flatpak-RenamedFrom

### PR DESCRIPTION
**Pull request by wjt**
Tested as follows:

* Installed Pidgin from Flathub (it has `X-Flatpak-RenamedFrom=pidgin.desktop;`)
* Manually edited the `icon-grid-layout` property to replace `org.libreoffice.LibreOffice-writer.desktop` with `libreoffice-writer.desktop` (this has `X-Endless-Alias`) and `im.pidgin.Pidgin.desktop` with `pidgin.desktop`
* Restarted the unmodified shell: Writer was still shown on the desktop but Pidgin disappeared
* Started the modified shell: both were shown on the desktop

https://phabricator.endlessm.com/T22596

### CODE REVIEW

**Comment by wjt** [comment]
Actually I think we need to write back the file with the alias swapped back – unless our modification to this key will survive Flatpak update?

File: src/shell-app-system.c [hunk 1]
```diff
@@ -113,13 +128,49 @@ add_aliases (ShellAppSystem  *self,
...
+   * be) ;-terminated, not just ;-delimited. This is so you can express an
+   * empty trailing element: "a;;" represents the list ["a", ""].
+   *
+   * It is possible to escape a semicolon in a list: "a\;b;" represents ["a;",
+   * "b"]. We cannot support this here, because g_key_file_get_string(), used
+   * internally by g_desktop_app_info_get_string(), rejects the "\;" sequence
+   * and returns NULL.
+   *
+   * https://gitlab.gnome.org/GNOME/glib/merge_requests/339
+   */
```
Comment (GeorgesStavracas) [inline src/shell-app-system.c]: This is a brilliant commit message, but not so much as a comment.

Comment (wjt) [reply src/shell-app-system.c]: I can move the bulk of it to the commit message, but I do really want to make a note of it in the code, in the hope that when we have GLib 2.60 with the function I wrote in that MR, we can update this code

**Comment by wjt** [comment]
Pushed a new version of the last patch which moves most of the mega-comment to the commit message, adds a few more notes in the commit message, and adds the missing trailing `;` to the example in the comment.

**Pull request merged**